test/unit/pdf/finance/invoice_generator_test.rb

```diff
@@ -40,5 +40,14 @@ class Pdf::Finance::InvoiceGeneratorTest < ActiveSupport::TestCase
 
     content = @generator.generate
     assert_not_nil content
+
+    # ensure an image is present in the PDF
+    assert_equal 1, content.scan(%r{/Type /XObject}).size
```

Comment:
how is image present in the PDF when we mock the image call?? `expects(:with_logo).yields(logo_file)`

Reply:
Well, we mock it with a real file:
```
logo_file = File.open(file_fixture('wide.jpg'), 'rb')
```

Reply:
got it, `yields`